### PR TITLE
Fix bug when too many layers in scene admin

### PR DESCRIPTION
**Pull request by jrmi**

### CODE REVIEW

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 1]
```diff
@@ -16,6 +20,49 @@ const sanitizeRestProps = ({
...
+  // Get first layer to know total
+  const { total, data: dataFirstPage } = await dataProvider(GET_LIST, RES_DATALAYER, {
+    pagination: { page: 1, perPage: PER_PAGE },
+    sort: { field: 'name', order: 'ASC' },
+    filter,
+  });
+
+  const pageCount = Math.ceil(total / PER_PAGE);
+
+  const result = Array.from({ length: pageCount });
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js]: Just a suggestion as I find this is more readable:

```suggestion
  const result = [...Array(pageCount)];
```

Another (not so good) alternative could be:

```suggestion
  const result = Array(pageCount).fill();
```

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 2]
```diff
@@ -16,6 +20,49 @@ const sanitizeRestProps = ({
...
+  // Keep first result
+  result[0] = dataFirstPage;
+
+  // Query all next layers
+  const promises = Array.from({ length: pageCount - 1 }).map(
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js]: As previous comment:

```suggestion
  const promises = [...Array(pageCount - 1)].map(
```

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 3]
```diff
@@ -16,6 +20,49 @@ const sanitizeRestProps = ({
...
+    (_, index) => dataProvider(GET_LIST, RES_DATALAYER, {
+      pagination: { page: index + 2, perPage: PER_PAGE },
+      sort: { field: 'name', order: 'ASC' },
+      filter,
+    }).then(({ data }) => {
+      result[index + 1] = data;
+    }),
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js]:
```suggestion
    async (_, index) => {
      const queryParams = {
        pagination: { page: index + 2, perPage: PER_PAGE },
        sort: { field: 'name', order: 'ASC' },
        filter,
      };

      const { data } = await dataProvider(GET_LIST, RES_DATALAYER, queryParams);
      result[index + 1] = data
    },
```

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 4]
```diff
@@ -16,6 +20,49 @@ const sanitizeRestProps = ({
...
+  );
+
+  await Promise.all(promises);
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js]: Another pattern suggestion for this whole process:

```jsx
const results = [
  ...dataFirstPage,
  ...(await Promise.all(promises),)
];
```

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 5]
```diff
@@ -91,11 +135,23 @@ const GeolayerSelect = ({ dataProvider, onChange, excludeIds = [], includeIds =
...
+          </NativeSelect>
+        )
+        : (
+          <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
+            {geolayers.map(({ id, name }) => (
+              <MenuItem key={id} value={id}>{name}({id})</MenuItem>
+            ))}
+          </Select>
+        )
+      }
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js]: Always prefer avoiding ternary operator for multiline JSX:

```suggestion
      {/* Do we have too many element to have perf issues ? */}
      {(geolayers.length > MAX_NON_NATIVE_SELECT) && (
          <NativeSelect onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
            {geolayers.map(({ id, name }) => (
              <option key={id} value={id}>{name}({id})</option>
            ))}
          </NativeSelect>
        )}

        {(geolayers.length <= MAX_NON_NATIVE_SELECT) && (
          <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
            {geolayers.map(({ id, name }) => (
              <MenuItem key={id} value={id}>{name}({id})</MenuItem>
            ))}
          </Select>
        )}
```

File: src/modules/RA/Scene/components/GeolayerSelect.js (hunk 1 above)
Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js]: Replaced by later suggestion.

File: src/modules/RA/Scene/components/GeolayerSelect.js (hunk 2 above)
Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js]: Done

File: src/modules/RA/Scene/components/GeolayerSelect.js (hunk 3 above)
Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js]: Done

File: src/modules/RA/Scene/components/GeolayerSelect.js (hunk 4 above)
Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js]: Done

File: src/modules/RA/Scene/components/GeolayerSelect.js (hunk 5 above)
Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js]: Done

File: src/modules/RA/Scene/components/GeolayerSelect.js [hunk 6]
```diff
@@ -91,11 +136,24 @@ const GeolayerSelect = ({ dataProvider, onChange, excludeIds = [], includeIds =
...
+      )}
+
+      { (geolayers.length <= MAX_NON_NATIVE_SELECT) && (
       <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
         {geolayers.map(({ id, name }) => (
-          <MenuItem key={id} value={id}>{name}</MenuItem>
+          <MenuItem key={id} value={id}>{name}({id})</MenuItem>
         ))}
       </Select>
+      )}
```
Comment (mabhub) [inline src/modules/RA/Scene/components/GeolayerSelect.js:155]: As both *selects* receive same props, you may avoid duplicating JSX blocks by using a variable for storing the right component to use:

```jsx
  const SelectComponent = geolayers.length > MAX_NON_NATIVE_SELECT
    ? { Main: NativeSelect, Item: 'option' }
    : { Main: Select, Item: MenuItem };

  /*  OR:

  const SelectComponent = {};
  if (geolayers.length > MAX_NON_NATIVE_SELECT) {
    LocalSelect.Main = NativeSelect;
    LocalSelect.Item = 'option';
  } else {
    LocalSelect.Main = Select;
    LocalSelect.Item = MenuItem;
  }

  */

  return (
    <FormControl fullWidth>
      <InputLabel htmlFor="geolayer">Couche</InputLabel>

      <SelectComponent.Main onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
        {geolayers.map(({ id, name }) => (
          <SelectComponent.Item key={id} value={id}>{name}({id})</SelectComponent.Item>
        ))}
      </SelectComponent.Main>
    </FormControl>
  );
```

Comment (jrmi) [reply src/modules/RA/Scene/components/GeolayerSelect.js:155]: May be later ;-) but thanks

**Pull request merged**